File: app/src/main/java/andiag/coru/es/welegends/rest/entities/persist/Image.java
package andiag.coru.es.welegends.rest.entities.persist;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by andyqm on 06/06/2016.
 */
public class Image {
    private String full;
    private String sprite;
    private String group;
    private Map<String, Object> additionalProperties = new HashMap<String, Object>();

    public String getFull() {
        return full;
    }

    public void setFull(String full) {
        this.full = full;
    }

    public String getSprite() {
        return sprite;
    }

    public void setSprite(String sprite) {
        this.sprite = sprite;
    }

    public String getGroup() {
        return group;
    }

    public void setGroup(String group) {
        this.group = group;
    }

    public Map<String, Object> getAdditionalProperties() {
        return this.additionalProperties;
    }

    public void setAdditionalProperty(String name, Object value) {
        this.additionalProperties.put(name, value);
    }

}

File: WeLegends/app/src/main/java/es/coru/andiag/welegends/common/entities/Image.kt
package es.coru.andiag.welegends.common.entities

import java.io.Serializable
import java.util.*

/**
 * Created by Canalejas on 08/12/2016.
 */

class Image : Serializable {

    var full: String? = null
    var sprite: String? = null
    var group: String? = null
    private var additionalProperties: MutableMap<String, Any>? = null

    fun getAdditionalProperties(): MutableMap<String, Any>? {
        return this.additionalProperties
    }

    fun setAdditionalProperties(additionalProperties: MutableMap<String, Any>) {
        this.additionalProperties = additionalProperties
    }

    fun addAdditionalProperty(name: String, value: Any) {
        if (this.additionalProperties == null) {
            this.additionalProperties = HashMap<String, Any>()
        }
        this.additionalProperties!!.put(name, value)
    }

}
